transfer_learning.py

```diff
@@ -0,0 +1,229 @@
+import torch
+import time
+import json
+import evaluate
+import argparse
+import numpy as np
+import pandas as pd
+from datasets import Dataset
+from transformers import AutoModelForImageClassification, Trainer, TrainingArguments
+# from torchvision import models
+
+
+class dataset():
+    '''
+    Class defining the creation of a huggingface Dataset for training. Probably doesn't
+    need to be a class rather than just a couple functions, but fuck it.
+    '''
+    def __init__(self, data_path='data/tensors.pt', label_path='data/anime_label_map.json', noise_path=None, train_test_split=0):
+        self.data_path = data_path
+        self.label_path = label_path
+        self.noise_path = noise_path
+        self.train_test_split=train_test_split
+
+    def get_dataset(self):
+        '''
+        Loads a pytorch tensor file at a path and returns a huggingface Dataset for
+        ease of use in fine-tuning a model.
+        '''
+        
+        # load all data, optionally with noise
+        tensors = torch.load(self.data_path)
+        if self.noise_path is not None:
```

Comment:
If I am reading this right, it combines the `tensors.pt` and `tensors_noise.pt` together into one dataset file? This shouldn't be necessary. `tensors_noise.pt` contains all tensors already in `tensors.pt`.

Reply:
oh nice, I didn't realize. changing rn